and-cli-dotnet-test.test.js

```diff
@@ -2,15 +2,125 @@
 // #region Imports
 // -----------------------------------------------------------------------------------------
 
-const { shouldDisplayHelpMenu } = require("./tests/describes");
+const { dotnetTest } = require("./modules/commands");
+const {
+    shouldDisplayError,
+    shouldDisplayHelpMenu,
+    withTemporaryDirectory,
+} = require("./tests/shared-specs");
+const dotnetPath = require("./modules/dotnet-path");
+const dotnetTestModule = require("./modules/dotnet-test");
+const testUtils = require("./tests/test-utils");
 
 // #endregion Imports
 
+// -----------------------------------------------------------------------------------------
+// #region Constants
+// -----------------------------------------------------------------------------------------
+
+const COMMAND = dotnetTest.command;
+const OPTIONS = dotnetTestModule.getOptions();
+const TEST_RUN_SUCCESSFUL_REGEXP = /Test Run Successful/g;
+
+// #endregion Constants
+
 // -----------------------------------------------------------------------------------------
 // #region Tests
 // -----------------------------------------------------------------------------------------
 
 describe("and-cli-dotnet-test", () => {
+    // -----------------------------------------------------------------------------------------
+    // #region Setup
+    // -----------------------------------------------------------------------------------------
+
+    withTemporaryDirectory(COMMAND, () => {
+        dotnetPath.verifyOrExit();
+    });
+
+    // #endregion Setup
+
+    // -----------------------------------------------------------------------------------------
+    // #region Run by project
+    // -----------------------------------------------------------------------------------------
+
+    describe("Run by project", () => {
+        describe("when no solution can be found", () =>
+            shouldDisplayError(
+                async () =>
+                    // Arrange & Act
+                    await testUtils.executeCliCommand(COMMAND, [
+                        OPTIONS.BY_PROJECT,
+                    ])
+            ));
+
+        describe("when solution exists", () => {
```

Comment:
I did play around with a 'test failure' scenario, but I ended up removing it. It was a little hacky - I had to do a find & replace in the generated unit test file to inject the failing test. I had it working locally, but all of the Travis builds were failing - I think there's something with the way the suite is executing child processes that needs to be worked out - it was holding onto a process and Jest was never exiting, so it would time out after 10-12 minutes of inactivity. 

Definitely a good idea, though, I appreciate the thought!